### ISSUE TITLE
Feature Request - Schema implementation for Header / Footer

### ISSUE THREAD

**Issue by dollhauss** [pojome/elementor#3990]
Hey guys,

I've been playing with the beta and I notice it does not include Schema meta tags for the header & footer.

This is essential for SEO.

At the moment the header / footer options are no different to creating a saved section template and hooking it in.

Even Astra Pro includes Schema in their header / footer Custom Layouts which make it a far more powerful solution than using the new Elementor Pro 2 option.

Will you be adding Schema?

Cheers.

**Comment by bainternet** [comment]
@dollhauss 

Not sure if its something we will implement, it's more of a 3rd party plugin territory.

Just to get a better understanding, what kind of schema are you missing for header and footer?

**Comment by dollhauss** [comment]
This should be implemented as part of the themer as without it there is nothing to tell Google that the section is a header / footer. A plugin should not have to achieve this.

For example, Beaver Themer adds the markup:
itemtype="http://schema.org/WPHeader"
to a custom header section, and also includes:
itemtype="https://schema.org/ImageObject"
for any images / logos within that section.

And for the footer:
itemtype="https://schema.org/WPFooter"

Even Astra Pro's Custom Layout extension adds the correct Schema to a header and footer field. Check it out.

This is extremely important if you are going to be replacing the header and footer.
If you don't include this you don't really have a header or a footer, you just have some hooked in global templates.

Hope this helps and you can look at implementing this!

Cheers.

**Comment by themedleb** [comment]
+1

**Comment by quantumleap33** [comment]
Guys, I was just building a full site with Elementor and the Hello theme and I run into this issue. Is there an ETA for this?

**Comment by black-eye** [comment]
+1

**Comment by fxaviers** [comment]
+1

**Comment by gabrieluntura** [comment]
+1
This is much needed

**Comment by mikesale** [comment]
+1
Why oh why?

**Comment by liorkesos** [comment]
Isn't this like a trivial thing? Just associating some markup to the header and footer?
For the people deeply passionate about this - issue a pull requests and reference the issue.
For people that aren't technical I'm pretty sure that of you invest 200$ you can sponser this development.
Iight be greatly oversimplifying this bit of it was important to me I would issue a pull requests - that is the true power of open source.

**Comment by arielk** [comment]
I hope it will be merged into one of the upcoming versions of **Elemental Pro**, probably in v.2.2 or v2.3

Thanks everyone!

**Comment by arielk** [comment]
Duplicate of #2385

**Comment by silvioghedin** [comment]
+1

**Comment by rcarser** [comment]
It isn't a good idea. Many users add Schema through Google Webmaster Tools. Other users do it through plugins. We also have to keep in mind that many themes incorporate Schema in core. Personally, I think this job belongs to third-party developers. Integrate Shema with Elementor could be cause of problems and incompatibility for some users.

**Comment by quantumleap33** [comment]
I disagree. 

How can users add Shema data to the Elementor elements? Even with plugins. All the plugins do not affect the header and footer elements. And do we need plugin after plugin to have this functionality? If this is so then, just have Elementor basic functionality and buy another 10 plugins for the rest with the possibility of one of them breaking on the next update! I believe schema meta tags should be added to the hello theme but most importantly inside all related Elementor elements since so many have started affecting theme functionality territory.

**Comment by rcarser** [comment]
I reaffirm myself. I think that it's a third-party territory. The Schema structure is very ample and complex and each project will need a certain Schema implementation. Personally, I think this work should be done by theme developers. It's very easy to add a Schema structure to any site created with Elementor Pro. If you are interested, I invite you to try the elementor hook '**elementor/widget/render_content**'. Through this hook you can modify all content generated with Elementor in the frontend and add the Shema meta tags. It is quite simple. I encourage you to try it!

**Comment by jaredgeddy** [comment]
This needs to happen at the widget level. It's not rocket science, just add an area to add a schema.org url on a widget. Just like custom css that gets injected inline except it's a schema.org url reference or property. There is currently no way for me to tell search engines what my content is beyond just having content.

For those who say this is a third party obligation, you are flat wrong. Why would a third party plugin manipulate elements of my page builder? That would fall under the page builder's responsibility. Just like building drag and drop elements that are built with HTML5 and not tables. Elementor should add this as a feature to build current SEO websites.

For those who are interested in seeing this feature for the sake of improving their websites they build using Elementor I've created [a community on gitter.im](https://gitter.im/Elementor-Wordpress-Plugin/Lobby?utm_source=share-link&utm_medium=link&utm_campaign=share-link) for collaborating on this project:

**Comment by LauGau** [comment]
> Just like custom css that gets injected inline except it's a schema.org url reference or property.

+1

**Comment by black-eye** [comment]
I'm baffled - was it implemented or not?
It doesn't seem so, at least not for a footer...

**Comment by shujainali** [comment]
It's been a one year but looks like schema feature is still not available :( 

![Schema](https://user-images.githubusercontent.com/59234099/71449706-7f151880-2774-11ea-93bb-0ae739774a7e.jpg)

**Comment by masoudahmadpour** [comment]
+1
